build_tools/build_libtorch.sh

```diff
@@ -102,19 +104,39 @@ fi
 
 checkout_pytorch() {
   if [[ ! -d "$PYTORCH_ROOT" ]]; then
-    git clone https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
+    git clone --depth 1 --single-branch --branch "${PYTORCH_BRANCH}" https://github.com/pytorch/pytorch "$PYTORCH_ROOT"
   fi
   cd "$PYTORCH_ROOT"
-  git fetch --all
-  git checkout origin/"${PYTORCH_BRANCH}"
-  git submodule update --init --recursive
+  for dep in protobuf pocketfft cpuinfo FP16 psimd fmt sleef pybind11 onnx flatbuffers foxi; do
```

Comment:
minimal number of third party deps (since we're turning off all build flags anyway)